### ISSUE TITLE
shapes inside svg tag in view mode

### ISSUE THREAD

**Issue by revyh** [jkphl/svg-sprite#95]
As I understand, in view mode for all shapes created <code>&lt;svg&gt;</code> wrapper and <code>&lt;view&gt;</code>. Then, when I try to refer this views in inline svg using the fragment identifier, it doesn't work. I tried it in FF and Chormium. At the same time, all works perfectly when I try to refer to the shape directly by its wrapper, not view. Also, everything is fine if I refer to the view in <code>&lt;img&gt;</code>.

When I tried to fix this, I noticed that in <a href="https://css-tricks.com/svg-fragment-identifiers-work/">Chris Coer's article</a> (which you mention in the docs) and other articles used by <code>&lt;g&gt;</code> wrapper for shapes rather than <code>&lt;svg&gt;</code>. And it works just as I imagined.

Did I miss the point? What the reason is to enclose all the shapes in <code>&lt;svg&gt;</code>?

Just in case, here is my config:
<pre>
  svgSpriteConfig = {
    shape         : {
      dimension   : {
        maxWidth  : 64,
        maxHeight : 64
      }
    },
    mode          : {
      view        : {
        dest      : '.',
        sprite    : 'icons.svg',
        bust      : false,
      }
    }
  };
</pre>

P.S. If anything, I'm sorry for my english :)

**Comment by jkphl** [comment]
Hey @revyh,

first of all, don't worry about your English! It's not worse than mine, for sure. ;)

As far as I know, there shouldn't be any difference between wrapping the single shapes in `<svg>` or `<g>` elements (in this regard at least). Have you tried replacing the `<svg>`s with `<g>`s`? Does it work then? I doubt it ...

Also, as far as I know, [support for fragment identifiers](http://caniuse.com/#feat=svg-fragment) on SVGs is still incomplete, buggy and inconsistent between browsers. I've got no information at hand whether fragment identifier should already work with inline SVGs (and which browsers do support it).

If possible, please provide a **working example** (like a CodePen / JSFiddle) of what you want to achieve and I'll see what I can do.

Cheers,
Joschi

**Comment by revyh** [comment]
Hi @jkphl! Thx for your quick reply (I can't boast the same).

**tl;dr** yeah, you right.

I just play with svg sprites. That's why I try to use this incomplete, buggy and inconsistent fragment identifier. Well... I think, that I've played enough with it in a last day:)

Whatever it was, what I wanted to achieve is to use an external sprite with code like this:
```html
<svg><use xlink:href="sprite.svg#icon_name"/></svg>
```
And it would be great if I could still use fragment identifiers in CSS background (for the icons in the input fields of forms), like this:
```css
background: url(sprite.svg#icon_name);
```
In articles (mentioned Coer's and others), I've seen the use of views with `<img>` and thought that it should work with the inline svg. Well, I was wrong. And I still can't get it. If some part of the svg technology works with `<img>`, then why the same part don't work with inline svg? Maybe some security restrictions or limitations on the standard? (Hard to believe that this feature just isn't implemented in modern browsers).

I tried to replace `<svg>`s with `<g>`s, but it didn't gave me any effect. (Actually, I found one difference between them, but it so small and impractical for current output modes, that I wouldn't describe it here). Eventually, I've created an [working example on codepen](http://codepen.io/revyh/pen/vOegQW) of the use of fragment identifiers with all output modes. Sadly, it doesn't work as on my localhost. I think this is due to the same-origin policy. So, instead of hosting assets on codepen, I just added a screenshot of my localhost to example.

**Comment by jkphl** [comment]
Thanks for all the details. So as this obviously isn't an issue if *svg-sprite*, I'll close this issue for now. Please feel free to reopen it in case of further problems.
Cheers, Joschi

**Comment by yairEO** [comment]
@jkphl - for people to be able to use svg fragments, the SCSS should be generated something like this:

    @mixin common($path){
	  background: url("../images/svg/sprite/view/sprite.svg#{'#' + $path }") center center no-repeat;
    }
  
    // an example for a single fragment:
    .svg-pie-chart {
	    @include common('pie-chart');
    }

**Comment by jkphl** [comment]
@yairEO You're free to use a customized version of the default SCSS rendering template which can be found in the `tmpl` directory (depending on the rendering mode you use). Copy it elsewhere, customize it appropriately and adapt your configure. That should do it. :)
